src/EditorFeatures/CSharpTest/Diagnostics/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            TestMissing(
+@"class Class { [|int i|]; int j; int P { get { return i; } set { j = value; } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldAndPropertyHaveDifferentStaticInstance()
+        {
+            TestMissing(
+@"class Class { [|static int i|]; int P { get { return i; } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldUseInRefArgument1()
+        {
+            TestMissing(
+@"class Class { [|int i|]; int P { get { return i; } } void M(ref int x) { M(ref i); } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldUseInRefArgument2()
+        {
+            TestMissing(
+@"class Class { [|int i|]; int P { get { return i; } } void M(ref int x) { M(ref this.i); } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldUseInOutArgument()
+        {
+            TestMissing(
+@"class Class { [|int i|]; int P { get { return i; } } void M(out x) { M(out i); } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestNotWithVirtualProperty()
+        {
+            TestMissing(
+@"class Class { [|int i|]; public virtual int P { get { return i; } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestNotWithConstField()
+        {
+            TestMissing(
+@"class Class { [|const int i|]; int P { get { return i; } } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldWithMultipleDeclarators1()
+        {
+            Test(
+@"class Class { int [|i|], j, k; int P { get { return i; } } }",
+@"class Class { int j, k; int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldWithMultipleDeclarators2()
+        {
+            Test(
+@"class Class { int i, [|j|], k; int P { get { return j; } } }",
+@"class Class { int i, k; int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldWithMultipleDeclarators3()
+        {
+            Test(
+@"class Class { int i, j, [|k|]; int P { get { return k; } } }",
+@"class Class { int i, j; int P { get; } }");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        public void TestFieldAndPropertyInDifferentParts()
+        {
+            Test(
+@"partial class Class { [|int i|]; } partial class Class { int P { get { return i; } } }",
+@"partial class Class { } partial class Class { int P { get; } }");
```

Comment:
you could use `<workspace><project><document>` literals to construct a workspace with 2 documents for this test. That gives you [coverage for this branch of code](https://github.com/CyrusNajmabadi/roslyn/blob/c43fd6469516e05b2c08df16a4c51d5f952deeab/src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyCodeFixProvider.cs#L81-L90)